youtube_dl/downloader/__init__.py

```diff
@@ -30,14 +32,17 @@ def get_suitable_downloader(info_dict, params={}):
     protocol = determine_protocol(info_dict)
     info_dict['protocol'] = protocol
 
+    if (info_dict.get('start_time') or info_dict.get('end_time')) and FFmpegFD.can_download(info_dict):
```

Comment:
I don't think so. If I explicitly specify an `--external-downloader` I do this intentionally and express the interntion to use exactly this extractor thus all the consequences (that the whole video will be downloaded instead of a fragment) are my responsibility.